api/envoy/config/core/v3/base.proto

```diff
@@ -313,6 +313,24 @@ message HeaderMap {
   repeated HeaderValue headers = 1;
 }
 
+// A path that is watched for changes, e.g. by inotify on Linux.
+message WatchedPath {
+  enum Event {
+    // Generated for the directory containing the new filename when a file is
+    // renamed.
+    MOVED_TO = 0;
+
+    // File was modified, e.g. write(2), truncate(2).
+    MODIFIED = 1;
+  }
+
+  // Path to watch.
```

Comment:
Can you beef this up to clarify that this should be a path to a file? Or is the intention to support directory watching also? Per @markdroth comment about maybe splitting the messages and increasing the docs would help?

Reply:
I wonder is the file modify variant necessary or could it be dropped in favor of directory move watch only? 

Capability of watching modify is of course nice for completeness, and thanks to the new hashing approach, it probably has a good chance on returning consistent files, but shouldn't symlink/directory swap still be recommended?  Also, the `WatcherImpl` for inotify will set watch for directory regardless of user asking for file modify events, so there is little difference between the two on Linux.

Reply:
Yep, back to WatchedDirectory.